common/common-api/src/main/java/org/ow2/proactive/http/CommonHttpResourceDownloader.java

```diff
@@ -0,0 +1,155 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.http;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+
+import org.apache.http.Header;
+import org.apache.http.HttpStatus;
+import org.apache.http.StatusLine;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.log4j.Logger;
+import org.ow2.proactive.utils.FileUtils;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 28/11/2017
+ */
+public class CommonHttpResourceDownloader {
+
+    private static final Logger logger = Logger.getLogger(CommonHttpResourceDownloader.class);
+
+    private static final Integer CONNECTION_POOL_SIZE = 5;
+
+    private static final String CONTENT_DISPOSITIOB_REGEXP = "(?i)^.*filename=\"([^\"]+)\".*$";
+
+    private static CommonHttpResourceDownloader instance = null;
+
+    private CloseableHttpClient client;
+
+    public static CommonHttpResourceDownloader getInstance() {
+        if (instance == null) {
+            instance = new CommonHttpResourceDownloader();
+        }
+        return instance;
+    }
+
+    private CommonHttpResourceDownloader() {
+
+    }
+
+    public String getResource(String sessionId, String url, boolean insecure) throws IOException {
+
+        CommonHttpClientBuilder builder = new CommonHttpClientBuilder().maxConnections(CONNECTION_POOL_SIZE)
+                                                                       .useSystemProperties()
+                                                                       .insecure(insecure);
+        try (CloseableHttpClient client = builder.build()) {
+            HttpGet request = new HttpGet(url);
+
+            if (sessionId != null) {
+                request.setHeader("sessionid", sessionId);
+            }
+            CloseableHttpResponse response = client.execute(request);
+
+            StatusLine status = response.getStatusLine();
+            if (status.getStatusCode() != HttpStatus.SC_OK) {
+                throw new IOException(String.format("Cannot access resource %s: code %d", url, status.getStatusCode()));
+            }
+
+            return readContent(response.getEntity().getContent());
+        }
+    }
+
+    public UrlContent getResourceContent(String sessionId, String url, boolean insecure) throws IOException {
+
+        CommonHttpClientBuilder builder = new CommonHttpClientBuilder().maxConnections(CONNECTION_POOL_SIZE)
```

Comment:
this first part is the same as the other method getResource(.... and could be extracted in a private method